src/axom/quest/examples/quest_distributed_distance_query_example.cpp

```diff
@@ -732,6 +766,128 @@ class QueryMeshWrapper
     }
   }
 
+  /**
+   * Return number of errors found on the local mesh partition.
+   * Populate "error_flag" field with the number of errors, for
+   * visualization.  Randomized circle points (--random-spacing
+   * switch) can cause false positives, so when on, distance
+   * inaccuracy is a warning (not an error) for the purpose of
+   * checking.
+   */
+  template <int NDIMS>
+  int checkClosestPoints(const Circle& circle, const Input& params)
+  {
+    using PointType = Circle::PointType;
+    using PointArray = axom::Array<PointType>;
+    using SegmentType = primal::Segment<double, NDIMS>;
+    using IndexSet = slam::PositionSet<>;
+
+    auto queryPts = getVertexPositions<PointArray>();
+
+    auto cpPositions =
+      getParticleMesh().getNodalVectorField<PointType>("closest_point");
+
+    auto cpIndices =
+      getParticleMesh().getNodalScalarField<axom::IndexType>("cp_index");
+
+    assert(queryPts.size() == cpPositions.size());
+    assert(queryPts.size() == cpIndices.size());
+
+    SLIC_INFO(axom::fmt::format("Closest points ({}):", cpPositions.size()));
+
+    /*
+      Allowable slack is half the arclength between 2 adjacent circle
+      points.  A query point on the circle can correctly have that
+      closest-distance, even though the analytical distance is zero.
+      If spacing is random, distance between adjacent points is not
+      predictable, leading to false positives.  We don't claim errors
+      for this in when using random.
+    */
+    const double avgObjectRes =
+      2 * M_PI * params.circleRadius / params.circlePoints;
+    const double allowableSlack = avgObjectRes / 2;
+
+    int sumErrCount = 0;
+    int sumWarningCount = 0;
+    auto& errorFlag = *getDC()->GetField("error_flag");
+    for(auto i : IndexSet(queryPts.size()))
+    {
+      const auto& qPt = queryPts[i];
```

Comment:
Suggestion: A few small comments to this function might help the reader.
E.g.
* `// check that points within threshold have a distance`
* `// check that computed closest point is near its corresponding closest point on the circle (within tolerance)`

Reply:
Will do.